apps/vmq_commons/include/vmq_types.hrl

```diff
@@ -8,19 +9,7 @@
 -define(true, 1).
 -define(false, 0).
 
--type peer()                :: {inet:ip_address(), inet:port_number()}.
--type username()            :: binary() | undefined.
--type password()            :: binary() | undefined.
--type client_id()           :: binary().
--type mountpoint()          :: string().
--type subscriber_id()       :: {mountpoint(), client_id()}.
--type reg_view()            :: atom().
--type topic()               :: [binary()] | binary().
--type qos()                 :: 0 | 1 | 2.
--type routing_key()         :: [binary()].
--type payload()             :: binary().
--type flag()                :: ?true | ?false | boolean() | empty. % empty for test purposes
--type msg_ref()             :: binary().
+-type fflag()               :: flag() | ?true | ?false | empty. % empty for test purposes
```

Comment:
So `flag()` (and all the other types) now lives in `vernemq_dev` and I see `flag()` is now a type alias for `boolean()`. Was the idea to make `flag()` a 'clean' flag type and then make `fflag()` the internal version we need which can take on a couple of other meanings/values as well? What does the prefixed `f` mean?